pkg/apis/camel/v1alpha1/types.go

```diff
@@ -61,3 +62,53 @@ const (
 	IntegrationPhaseRunning   IntegrationPhase = "Running"
 	IntegrationPhaseError     IntegrationPhase = "Error"
 )
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type IntegrationContextList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []IntegrationContext `json:"items"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type IntegrationContext struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              IntegrationContextSpec   `json:"spec"`
+	Status            IntegrationContextStatus `json:"status,omitempty"`
+}
+
+type IntegrationContextSpec struct {
+	Dependencies []string          `json:"dependencies,omitempty"`
+	Properties   []PropertySpec    `json:"properties,omitempty"`
+	Environment  []EnvironmentSpec `json:"environment,omitempty"`
+}
+
+type PropertySpec struct {
+	Name  string
+	Value string
+}
+type EnvironmentSpec struct {
+	Name  string
+	Value string
+}
+
+type IntegrationContextStatus struct {
+	Phase  IntegrationContextPhase `json:"phase,omitempty"`
+	Digest string                  `json:"digest,omitempty"`
+	Image  string                  `json:"image,omitempty"`
+	From   int                     `json:"from,omitempty"`
+}
+
+type IntegrationContextPhase string
+
+const (
+	// IntegrationContextPhaseEditing --
+	IntegrationContextPhaseEditing IntegrationContextPhase = "Editing"
```

Comment:
Is this a real status? Like "Draft"?

Reply:
Yes, renamed it to draft to make it more clear